app/src/main/java/com/claraberriel/mvpparking/mvp/presenter/ReleasePresenter.java

```diff
@@ -0,0 +1,74 @@
+package com.claraberriel.mvpparking.mvp.presenter;
+
+import android.util.Log;
+
+import androidx.annotation.VisibleForTesting;
+
+import com.claraberriel.mvpparking.entities.Parking;
+import com.claraberriel.mvpparking.mvp.model.ReleaseModel;
+import com.claraberriel.mvpparking.mvp.view.ReleaseView;
+
+public class ReleasePresenter {
+
+    private final ReleaseModel releaseModel;
+    private final ReleaseView releaseView;
+
+    public ReleasePresenter(ReleaseModel releaseModel, ReleaseView releaseView) {
+        this.releaseModel = releaseModel;
+        this.releaseView = releaseView;
+    }
+
+    public boolean onRelease() {
+        try {
+            releaseModel.checkIfAnyReservationExists();
+            if (isParkingLotNumberValid() && isSecurityCodeValid()) {
+                if (releaseModel.parkingRelease(releaseModel.getParkingLotNumber(releaseView.getParkingLotNumber()),
+                        releaseView.getSecurityCode())) {
+                    releaseView.showReservationReleased();
+                    return true;
+                } else {
+                    releaseView.showErrorParkingNumberSecurityCodeNotAMatch();
+                    return false;
+                }
+            }
```

Comment:
remember to only use **try** on the lines that could raise an exception

Reply:
Would it work like this:
`    public boolean onRelease() {
        try {
            releaseModel.checkIfAnyReservationExists();
        } catch (IllegalArgumentException e) {
            releaseView.showErrorNoExistingReservations();
            return false;
        }
        if (releaseModel.checkIfAnyReservationExists()) {
            if (isParkingLotNumberValid() && isSecurityCodeValid()) {
                if (releaseModel.parkingRelease(releaseModel.getParkingLotNumber(releaseView.getParkingLotNumber()),
                        releaseView.getSecurityCode())) {
                    releaseView.showReservationReleased();
                    return true;
                } else {
                    releaseView.showErrorParkingNumberSecurityCodeNotAMatch();
                    return false;
                }
            }
        }
        return false;
    }`

Reply:
Would it work like this:
`    public boolean onRelease() {

        try {

            releaseModel.checkIfAnyReservationExists();
        } catch (IllegalArgumentException e) {

            releaseView.showErrorNoExistingReservations();

            return false;

        }

        if (releaseModel.checkIfAnyReservationExists()) {

            if (isParkingLotNumberValid() && isSecurityCodeValid()) {

                if (releaseModel.parkingRelease(releaseModel.getParkingLotNumber(releaseView.getParkingLotNumber()),

                        releaseView.getSecurityCode())) {

                    releaseView.showReservationReleased();

                    return true;

                } else {

                    releaseView.showErrorParkingNumberSecurityCodeNotAMatch();

                    return false;

                }

            }

        }

        return false;

    }`